Sources/Core/UploadClient/UploadClient+Live.swift

```diff
@@ -63,20 +67,37 @@ extension UploadClient {
     private func upload(testData: TestResults) async throws {
       self.logger?.debug("Uploading \(testData)")
 
+      var data: Data
+      var response: HTTPURLResponse
       do {
-        let (data, _) = try await self.api.data(for: .upload(testData))
-        guard let result = try? self.api.decode(data, as: UploadResponse.self) else {
-          if let errorMessage = try? self.api.decode(data, as: UploadFailureResponse.self) {
-            throw UploadError.error(message: errorMessage.message)
-          } else {
-            throw UploadError.unknown
-          }
-        }
-        self.logger?.debug("Upload finished - \(result)")
+        (data, response) = try await self.api.data(for: .upload(testData))
       } catch {
-        self.logger?.error("Upload failed - \(error.localizedDescription)")
+        self.logger?.error("Error uploading: \(error.localizedDescription)")
         throw error
       }
+
+      // Ideally “HTTP 200 OK” etc, but maybe actually “HTTP 200 no error” etc.
+      let statusString = "HTTP \(response.statusCode) \(HTTPURLResponse.localizedString(forStatusCode: response.statusCode))"
+
+      // Currently this should get HTTP 202 Accepted, but let's be a bit permissive to future changes.
+      guard (201...204).contains(response.statusCode) else {
```

Comment:
New to Swift: Should we allow 200 OK response?

Reply:
I was thinking **no**… but I'm not entirely sure. I wanted to ensure we got a “something happened” response, ideally “201 Created” or “202 Accepted”. I extended it to 204 No Content for vague reasons that probably don't hold water. But mostly I wanted to be sure it's not just a “200 OK here's a random webpage” response.

![default web server page](https://github.com/user-attachments/assets/33299f8b-94b9-4645-8110-409f77f72fcb)

☝🏼 _do not want_